Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -414,6 +419,13 @@ class BrowserViewController: UIViewController,
         // Re-show toolbar which might have been hidden during scrolling (prior to app moving into the background)
         scrollController.showToolbars(animated: false)
 
+        browserDidBecomeActive()
+    }
+
+    func browserDidBecomeActive() {
```

Comment:
I think with this change we can remove the call to `startAtHomeCheck` under `tabManagerDidRestoreTabs` which was added in #16805, to patch that the start at home wasn't called at start up time.

Reply:
Thanks @lmarceau. It looks like that startAtHomeCheck is now storing the result into `didStartAtHome` which is then checked in BVC in `urlBarPresentCFR`:

```swift
// Checks if the app started with the Homepage tab
        // If true, don't show the CFR to avoid showing it on the Homepage
        guard !didStartAtHome else {
            didStartAtHome = false
            return
        }
```

At a glance it looks like I should be able to simply move that state (`didStartAtHome`) directly into BVC with this refactor, but I will take a closer look at how these are working together.

Reply:
@lmarceau Made this change in e06b285a7c3dc62c41415be4b84d3f7625f99af2. @PARAIPAN9 It looks like `didStartAtHome` was added for [FXIOS-7555](https://mozilla-hub.atlassian.net/browse/FXIOS-7555), can you LMK the best way to regression test that so I can be sure these refactors aren't impacting any of the CFR behaviors? Thank you

Reply:
Yeah they had to patch something related to the CFR. There's also another fix related to the same CFR in a PR opened [here](https://github.com/mozilla-mobile/firefox-ios/pull/17172). So I'll tag @PARAIPAN9 here for review. Whatever we merge first will affect the other PR.

Reply:
Yeah they had to patch something related to the CFR. There's also another fix related to the same CFR in a PR opened [here](https://github.com/mozilla-mobile/firefox-ios/pull/17095). So I'll tag @PARAIPAN9 here for review. Whatever we merge first will affect the other PR. 

Edit: I tagged the wrong PR

Reply:
Mhh. Thinking, since #17095 is needed for 120, better to merge that one first then we can adjust the behavior for v121 and adjust where we can 🤔

Reply:
> Thinking, since https://github.com/mozilla-mobile/firefox-ios/pull/17095 is needed for 120, better to merge that one first

@lmarceau @PARAIPAN9 I've added a Do Not Merge label for now, will wait until #17095 is merged in and then I can update this refactor to accommodate those changes.

Reply:
I merged it 👍 Let's work to improve this area. Let me know if I can do anything to help

Reply:
@lmarceau @PARAIPAN9 Update: I've rebased and updated this to incorporate Sorin's changes from #17095. Any extra 👀 are appreciated (I don't have a ton of context on the CFR fixes).

Reply:
I don't have full context either, so tagging Sorin on the PR for review on those changes.

Reply:
Just switch to this branch, and started reviewing it.

Reply:
I've test it without didStartAtHome check and it worked. I think it should be good.